src/net_processing.cpp

```diff
@@ -2489,19 +2489,10 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
             // We skip these operations for BLOCK_RELAY peers to avoid
             // potentially leaking information about our BLOCK_RELAY
             // connections via the addrman or address relay.
-            if (fListen && !::ChainstateActive().IsInitialBlockDownload())
-            {
-                CAddress addr = GetLocalAddress(&pfrom.addr, pfrom.GetLocalServices());
-                FastRandomContext insecure_rand;
-                if (addr.IsRoutable())
-                {
-                    LogPrint(BCLog::NET, "ProcessMessages: advertising address %s\n", addr.ToString());
-                    pfrom.PushAddress(addr, insecure_rand);
-                } else if (IsPeerAddrLocalGood(&pfrom)) {
-                    addr.SetIP(addrMe);
-                    LogPrint(BCLog::NET, "ProcessMessages: advertising address %s\n", addr.ToString());
-                    pfrom.PushAddress(addr, insecure_rand);
-                }
+            if (fListen && !::ChainstateActive().IsInitialBlockDownload()) {
+                AdvertiseLocal(pfrom);
```

Comment:
A question I had during last week's review club meeting:
Since this PR unifies the two spots from which self-advertisements are sent (`ProcessMessage` -> VERSION only to outbound peers, and `SendMessages` also to inbound peers) - Does this call during VERSION processing still have a point? 
If we simply removed it, wouldn't we self-advertise a few moment later anyway after we are successfully connected (because `m_next_local_addr_send` is initialized to 0 in the `SendMessages()` loop)?

Reply:
In commit "Unify self-announcement behavior".

There is no perfect overlap between the two mechanisms that trigger announcing. This one only sends to outbound connections, but does so even during IBD. The other periodic one also sends to inbound connections, but not during IBD.

I don't know if there is a point to keeping this one.

Reply:
Yeah, I also don't have a good answer here.
As Pieter notices, it's not *that* pointless, because of the send-during-IBD property. Now, whether that's something we want is an open question and requires understanding the expectations from addr relay as a whole.
I think this could be discussed/implemented separately in another PR.

Reply:
Actually, both of them are non-IBD? Both are inside the `!::ChainstateActive().IsInitialBlockDownload()` check?
So really, the periodic one should encompass the initial one.

The only reason to keep the initial one is if it somehow worked when the other one is not reached. But it's not true in practice, because the periodic happens *right before* actually sending ADDR message out.

I think the initial one should be dropped.